arviz/data/inference_data.py

```diff
@@ -252,6 +252,43 @@ def concat(*args, dim=None, copy=True, inplace=False, reset_dim=True):
     InferenceData
         A new InferenceData object by default.
         When `inplace==True` merge args to first arg and return `None`
+
+    Examples
+    --------
+    Use ``concat`` method to concatenate InferenceData objects. This will concatenates over
+    unique groups by default. We first create an InferenceData object:
+
+    .. ipython::
+
+        In [1]: import arviz as az
+           ...: import numpy as np
+           ...: data = {
+           ...:         "a": (["chain", "draw", "a_dim"], np.random.normal(size=(4, 100, 3))),
+           ...:         "b": (["chain", "draw"], np.random.normal(size=(4, 100))),
+           ...:         }
+           ...: coords = {
+           ...:         "chain": (["chain"], np.arange(4)),
+           ...:         "draw": (["draw"], np.arange(100)),
+           ...:         "a_dim": (["a_dim"], ["x", "y", "z"]),
+           ...:         }
+           ...: dataA = az.from_dict(posterior=data, coords=coords)
```

Comment:
I tried this code locally and it does not create an InferenceData object properly. ArviZ idiom should be followed:

```
data = { 
   "a": np.random.normal(size=(4, 100, 3)),
   "b": np.random.normal(size=(4, 100))
}
coords = {"a_dim": ["x", "y", "z"]}
dataA = az.from_dict(data, coords=coords, dims={"a": ["a_dim"]})
```

Reply:
Thanks, for the correction. It seems the xarray dataset (used in the previous case) map dims automatically, but we need to specify the mapping in case of `from_dict`. Is this the case?

Reply:
xarray's Dataset allows several notations, in the one used, the data and coordinates are specified as dicts of tuples, where the tuple contains both the data and the dimensions. ArviZ uses only one syntax, dicts of arrays for all 3 of them, the values of the dicts are directly understood as data, coordinate values or dimension list

Reply:
Okay, thanks for the clarification :)